src/Kafka/FFI.purs

```diff
@@ -0,0 +1,131 @@
+module Kafka.FFI
+  ( class RemoveKeysRL
+  , removeKeysRL
+  , Object
+  , objectFromRecord
+  , objectToRecord
+  ) where
+
+import Prelude
+
+import Control.Monad.ST as Control.Monad.ST
+import Foreign.Object as Foreign.Object
+import Foreign.Object.ST as Foreign.Object.ST
+import Option as Option
+import Prim.RowList as Prim.RowList
+import Type.Prelude as Type.Prelude
+import Unsafe.Coerce as Unsafe.Coerce
+
+-- | A version of `Option.Record _ _` that has the same underlying
+-- | representation as `Option.Option _` which can be used directly
+-- | in FFI definitions to represent JS Object with fields that may
+-- | or may not exist.
+newtype Object (required :: Row Type) (optional :: Row Type) =
```

Comment:
Could we add some documentation here explaining why we can't just use `Option.Record`? Maybe something like:
```
-- | We canot use `Option.Record` here because its FFI representation is not a plain JS object.
```

Reply:
That's a good point. I'll include it.